### ISSUE TITLE
Add unique user id to API

### ISSUE THREAD

**Issue by jjkester** [Inter-Actief/amelie#221]
Currently there is no method to request a unique identifier for a user using the API. This poses some problems with OAuth, since there is no proper way of matching a user to a local user when they log in using OAuth.

**Proposed fix:**

Add a field `id` to the `getPersonDetails` method.

**Comment by SebastiaandenBoer** [comment]
Currently, the method already provides the caller with a couple of reasonable identifiers:

- `studentId` (although this can be null)
- `employeeId` (although this can be null)
- `email`

Also, I fail to see the need for an ID needed for matching as this method can only be called for the currently authenticated user, meaning that you can always be assured that the result obtained from `getPersonDetails` is for the person currently authenticated.
This result can then be matched to the local user in the OAuth application.

**Comment by jjkester** [comment]
As you say yourself, the student- and employee identifiers can be null. The e-mail address is not to be trusted since I can change this in my profile. This would (in the very best case) lead to incorrectly identifying a user as a "new" user instead of a returning one.

For cases where someone authenticates only once (for example the Inter-Actief mobile app) the current situation is fine, but for web platforms having to match an Inter-Actief user to a local user a real unique and consistent identifier (in whatever size or shape) is necessary.

(Nearly) every OAuth provider provides a way of obtaining the ID. For example, GitHub has a special method for this that only returns the ID (so requesting the person details scope is not necessary).

TL;DR: I do not agree with you, only really unique and consistent identifiers work when having to match users on multiple login attempts.

**Comment by SebastiaandenBoer** [comment]
Yeah, after some discussion with @windwarrior I just realized that the situation where a person logs in multiple times could create certain issues. I'll get started on this right away.

**Comment by jjkester** [comment]
Great to hear! I would have stressed that no haste is to be made since for the symposium website we have at least a couple of months to spare, but I already see a pull request :smile:

**Comment by jjkester** [comment]
The only remark I have, which is an issue with the fix I proposed, is that by implementing an id on `getPersonDetails` the `account` scope is needed, providing access to a users account details.

Maybe, for privacy reasons, an additional method (with no required scope) might be a better way to do this.

**Comment by SebastiaandenBoer** [comment]
This should now be fixed in the pull request

**Comment by jjkester** [comment]
Awesome!
